src/sema/expression/member_access.rs

```diff
@@ -635,6 +648,67 @@ fn enum_value(
     }
 }
 
+fn event_selector(
+    loc: &pt::Loc,
+    expr: &pt::Expression,
+    id: &pt::Identifier,
+    file_no: usize,
+    contract_no: Option<usize>,
+    ns: &mut Namespace,
+    diagnostics: &mut Diagnostics,
+) -> Result<Option<Expression>, ()> {
+    if id.name != "selector" {
+        return Ok(None);
+    }
+
+    if let Ok(events) = ns.resolve_event(file_no, contract_no, expr, &mut Diagnostics::default()) {
+        match events.len() {
+            0 => Ok(None),
```

Comment:
Not tested.

Reply:
Not important. `ns.resolve_events()` will not return `Ok()` with zero events.

Reply:
I've removed the line